### PR TITLE
Add automatic preview workflows

### PR DESCRIPTION
**Pull request by awharn**
Adds workflows for automatic web-help previews for doc review. Since the workflows use pull_request_target, they will not run on this PR. :(

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -117,3 +119,94 @@ jobs:
...
+    - name: Resolve Artifact
+      run: |
+        mv docs/${{ github.ref_name }}/generatedWebHelp/* docs/${{ github.ref_name }}/.
+        rmdir docs/${{ github.ref_name }}/generatedWebHelp
+        rm docs/${{ github.ref_name }}/zowe.pdf
+        git add docs/${{ github.ref_name }}
+        git commit -sm "Add Zowe Web Help for ${{ github.ref_name }}"
+        git push origin HEAD
+
+  deploy-pr:
```
Comment (zFernand0) [inline .github/workflows/main.yml:178]: This seems very similar to the `deploy-artifact`
It does have some very specific changes though (GH event, and GH `ref_name > event.number`)

Curious if it's possible to reuse some code here.
Not a big deal, so feel free to ignore if it's too complicated 😋

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -117,3 +119,94 @@ jobs:
...
+        gh release download ${{ github.event.inputs.zowe-version }} -p zowe-${{ github.event.inputs.zowe-version }}-WebHelp.zip
+        rm -rf docs/${{ github.event.inputs.zowe-version }}
+        mkdir -p docs/${{ github.event.inputs.zowe-version }}
+        unzip -nqq zowe-${{ github.event.inputs.zowe-version }}-WebHelp.zip -d docs/${{ github.event.inputs.zowe-version }}
+        mv docs/${{ github.event.inputs.zowe-version }}/zowe-${{ github.event.inputs.zowe-version }}-WebHelp/* docs/${{ github.event.inputs.zowe-version }}/.
+        rmdir docs/${{ github.event.inputs.zowe-version }}/zowe-${{ github.event.inputs.zowe-version }}-WebHelp/
+        rm zowe-${{ github.event.inputs.zowe-version }}-WebHelp.zip
+        git add docs/${{ github.event.inputs.zowe-version }}
+        git commit -sm "Add Zowe Web Help ${{ github.event.inputs.zowe-version }}"
+        git push origin HEAD
```
Comment (zFernand0) [inline .github/workflows/main.yml:146]: 💀

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
Really cool automation! 🥳 
- https://zowe.github.io/zowe-cli-web-help-generator/

File: .github/workflows/main.yml (hunk 1 above)
Comment (awharn) [reply .github/workflows/main.yml:178]: I couldn't think of a good way of pulling that off, unfortunately. We can look into it more during innovation sometime to see if we can clean this up. For now though, I think it should work. 😄

**Pull request merged**